pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -53,6 +53,22 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 		rs.TimeoutSeconds = ptr.Int64(cfg.Defaults.RevisionTimeoutSeconds)
 	}
 
+	// Default IdleTimeoutSeconds only in case we have a non-zero default and the latter is not larger than the revision timeout.
+	// A zero default or a zero value set from the user or a nil value skips timer setup at the QP side.
+	if rs.IdleTimeoutSeconds == nil {
+		if cfg.Defaults.RevisionIdleTimeoutSeconds < *rs.TimeoutSeconds && cfg.Defaults.RevisionIdleTimeoutSeconds != 0 {
+			rs.IdleTimeoutSeconds = ptr.Int64(cfg.Defaults.RevisionIdleTimeoutSeconds)
+		}
+	}
+
+	// Default ResponseStartTimeoutSeconds only in case we have a non-zero default and the latter is not larger than the revision timeout.
+	// A zero default or a zero value set from the user or a nil value skips timer setup at the QP side.
+	if rs.ResponseStartTimeoutSeconds == nil {
+		if cfg.Defaults.RevisionRequestStartTimeoutSeconds < *rs.TimeoutSeconds && cfg.Defaults.RevisionRequestStartTimeoutSeconds != 0 {
```

Comment:
`revision-response-start-timeout-seconds` is used to initialize RevisionRequestStartTimeoutSeconds so we use that.

```
	// We default this to what the user has specified
	nc.RevisionRequestStartTimeoutSeconds = nc.RevisionTimeoutSeconds

	if err := cm.Parse(data,
		cm.AsInt64("revision-response-start-timeout-seconds", &nc.RevisionRequestStartTimeoutSeconds),
	); err != nil {
		return nil, err
	}
```

Reply:
Can we update `RevisionRequestStartTimeoutSeconds` to match the key name in a subsequent PR?